src/components/svelte-web.ts

```diff
@@ -242,27 +239,16 @@ export default function registerWebComponent(
       this[prop] = boolProperties.has(prop) ? newValue !== null : newValue
     }
 
-    addEventListener(event: string, callback: Callback) {
-      if (!this.listenerRemovers.has(event)) {
-        this.listenerRemovers.set(event, new Map())
-      }
-
-      // Note: $on(<event>, callback) returns a function which removes the
-      // callback from the event.
-      const remover = this.component.$on(event, callback)
-
-      // We store the remover so we can look it up in |removeEventListener|.
-      this.listenerRemovers.get(event).set(callback, remover)
-
-      // TODO: We could do this but we don't know if the event is handled
-      // by the component or not so we could end up triggering the event
-      // twice (i.e. in the case of 'click')
-      // super.addEventListener(event, callback, options)
+    #ensureEventForwarder(type: string) {
+      // Make sure we don't create the eventForwarder multiple times.
+      if (this.#forwardEvents.has(type)) return;
+      this.component.$on(type, e => this.dispatchEvent(e))
     }
 
-    removeEventListener(event: string, callback: Callback) {
-      this.listenerRemovers.get(event)?.get(callback)?.()
-      this.listenerRemovers.get(event)?.delete(callback)
+    addEventListener(event: string, callback: EventListenerOrEventListenerObject, options?: boolean | AddEventListenerOptions) {
+      // Make sure the event is forwarded from the internal component.
+      this.#ensureEventForwarder(event)
+      super.addEventListener(event, callback, options)
```

Comment:
Unfortunately for HTML events we override this means we get a double dispatch. I.e. for our click events we'll get a `PointerEvent` from the browser and a `CustomEvent` from our `LeoButton`